router/xgress_edge/listener.go

```diff
@@ -153,7 +153,10 @@ func (proxy *ingressProxy) processConnect(req *channel2.Message, ch channel2.Cha
 	// fabric connect
 	log.Debug("dialing fabric")
 	peerData := make(map[uint32][]byte)
-	peerData[edge.PublicKeyHeader] = req.Headers[edge.PublicKeyHeader]
+
+	if pk, found := req.Headers[edge.PublicKeyHeader]; found {
```

Comment:
I found this too, but wasn't able to confirm the fix. I couldn't reproduce the situation.

I also found that sometimes the header was being sent in w/ a length of zero.

Reply:
My current version was:

```
	if pk, found := req.Headers[edge.PublicKeyHeader]; found && len(pk) > 0{
		peerData[edge.PublicKeyHeader] = pk
	}
```

Reply:
But it didn't feel right. The header shouldn't be sent if it is going to be empty.